cmd/pulumi-converter-yaml/main.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// pulumi-converter-yaml is the "language converter" for Pulumi programs written in YAML or JSON. It is
+// responsible for translating JSON/YAML templates into PCL.
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"path/filepath"
+
+	yamlgen "github.com/pulumi/pulumi-yaml/pkg/pulumiyaml/codegen"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/pcl"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/encoding"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/rpcutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	"github.com/spf13/afero"
+	"google.golang.org/grpc"
+)
+
+type yamlConverter struct {
+}
+
+func (*yamlConverter) Close() error {
+	return nil
+}
+
+func (*yamlConverter) ConvertState(ctx context.Context,
+	req *plugin.ConvertStateRequest,
+) (*plugin.ConvertStateResponse, error) {
+	return nil, errors.New("not implemented")
+}
+
+// writeProgram writes a project and pcl program to the given directory
+func writeProgram(directory string, proj *workspace.Project, program *pcl.Program) error {
+	fs := afero.NewOsFs()
+	err := program.WriteSource(afero.NewBasePathFs(fs, directory))
```

Comment:
minor: This function should take the file system as an argument instead of building its own.
It can probably take just the fs, not the directory; letting the caller handle creating the base path FS.

```go
func writeProgram(fs afero.Fs, proj *workspace.Project, program *pcl.Program) error
```